src/components/views/rooms/EventTile.tsx

```diff
@@ -262,6 +264,10 @@ interface IState {
 
     thread: Thread | null;
     threadNotification?: NotificationCountType;
+    /**
+     * Whether the event is encrypted.
+     */
+    isRoomEncrypted: boolean;
```

Comment:
Would we not inherit this from the RoomView via a RoomContext instead of every single component calling the same async function possibly causing lots of rust overhead?

Reply:
Hmmm, i didn't aware of the room context shared across this component and others. I'll take a look to do it

Reply:
Its just a case of adding it to RoomView's state then all components can access it via RoomContext

Reply:
A pain point for class component is that you can't use multiple contexts directly. Some of them are already using the matrix client context.

Btw, I started to add it to the room view https://github.com/element-hq/element-web/pull/28278 few weeks ago, I'll continue the other PR